### PR TITLE
Adding prototype for Sheath Potential calculation [sheath-potential-dev]

### PR DESCRIPTION
**Pull request by mlstowell**
This PR demonstrates the type of solve that needs to be performed to compute the sheath potential on the surface of a mesh.

Once the sheath impedance is incorporated into the code we can compute the actual sheath potential.  This prototype uses a sheath impedance of 1.  The plan is to compute the electric field using a nested iteration:

0: Assume sheath potential is zero.
1: Compute E using -GradPhi as a Dirichlet BC
2: Compute D = epsilon E (solve an H(Div) mass matrix)
3. phi = z(phi, z_previous) n.D, repeat until z converges (solve an H1 mass matrix in each iteration) 
4. Go back to 1 unless the change in GradPhi is below some tolerance.

### CODE REVIEW

**Comment by v-dobrev** [comment]
@mlstowell, it is better to reserve the `ready-for-review` label just for PRs into `master`. Maybe we can add another label for PRs into other branches.

File: miniapps/plasma/cold_plasma_dielectric_solver.cpp [hunk 1]
```diff
@@ -291,6 +290,7 @@ CPDSolver::CPDSolver(ParMesh & pmesh, int order, double omega,
      e_b_(NULL),
      e_v_(NULL),
      d_v_(NULL),
+     phi_v_(NULL),
```
Comment (stefanhenneking) [inline miniapps/plasma/cold_plasma_dielectric_solver.cpp:293]: Missing initialization `phi_(NULL)` ?

File: miniapps/plasma/cold_plasma_dielectric_solver.cpp [hunk 2]
```diff
@@ -359,7 +356,7 @@ CPDSolver::CPDSolver(ParMesh & pmesh, int order, double omega,
    // Define compatible parallel finite element spaces on the parallel
    // mesh. Here we use arbitrary order H1, Nedelec, and Raviart-Thomas finite
    // elements.
-   // H1FESpace_    = new H1_ParFESpace(pmesh_,order,pmesh_->Dimension());
+   H1FESpace_    = new H1_ParFESpace(pmesh_,order,pmesh_->Dimension());
```
Comment (stefanhenneking) [inline miniapps/plasma/cold_plasma_dielectric_solver.cpp:359]: I was wondering about the order of approximation here. To be consistent with the first exact sequence spaces, should the RT space use `order-1` ?

File: miniapps/plasma/cold_plasma_dielectric_solver.cpp [hunk 3]
```diff
@@ -373,12 +370,20 @@ CPDSolver::CPDSolver(ParMesh & pmesh, int order, double omega,
    }
    if (kCoef_)
    {
+      if (L2FESpace_ == NULL)
+      {
+         L2FESpace_ = new L2_ParFESpace(pmesh_,order-1,pmesh_->Dimension());
+      }
```
Comment (stefanhenneking) [inline miniapps/plasma/cold_plasma_dielectric_solver.cpp:376]: Here, `order-1` is used for the L2 space, but the L2 vector space below uses `order`. 
Should it both be `order-1`?

File: miniapps/plasma/cold_plasma_dielectric_solver.cpp [hunk 4]
```diff
@@ -714,6 +719,7 @@ CPDSolver::~CPDSolver()
    // delete e_i_;
    delete e_;
    delete d_;
+   delete phi_;
```
Comment (stefanhenneking) [inline miniapps/plasma/cold_plasma_dielectric_solver.cpp:722]: Missing `delete phi_v_` ?

**Pull request closed without merging**